src/engine/source/kvdbstore/test/src/unit/kvdb_manager_test.cpp

```diff
@@ -1,7 +1,218 @@
+#include <memory>
+#include <string>
+
 #include <gtest/gtest.h>
 
-TEST(KVDBManagerTest, init)
+#include <cmstore/icmstore.hpp>
+
+#include <kvdb/kvdbManager.hpp>
+
+namespace
 {
-    // Dummy test to verify that KVDBManager can be instantiated.
-    GTEST_SKIP() << "Not implemented yet.";
+
+class BaseFakeNS : public cm::store::ICMStoreNSReader
+{
+public:
+    explicit BaseFakeNS(std::string ns)
+        : m_ns(std::move(ns))
+    {
+    }
+
+    // ICMStoreNSReader
+    const cm::store::NamespaceId& getNamespaceId() const override { return m_ns; }
+    cm::store::dataType::Policy getPolicy() const override { throw std::logic_error("unused"); }
+    cm::store::dataType::Integration getIntegrationByName(const std::string&) const override
+    {
+        throw std::logic_error("unused");
+    }
+    cm::store::dataType::Integration getIntegrationByUUID(const std::string&) const override
+    {
+        throw std::logic_error("unused");
+    }
+    bool integrationExistsByName(const std::string&) const override { return false; }
+    bool integrationExistsByUUID(const std::string&) const override { return false; }
+    virtual json::Json getKVDBByName(const std::string&) const override = 0;
+    json::Json getKVDBByUUID(const std::string&) const override { throw std::logic_error("unused"); }
+    bool kvdbExistsByName(const base::Name&) const override { return true; }
+    bool kvdbExistsByUUID(const std::string&) const override { return false; }
+    json::Json getAssetByName(const base::Name&) const override { throw std::logic_error("unused"); }
+    json::Json getAssetByUUID(const std::string&) const override { throw std::logic_error("unused"); }
+    bool assetExistsByName(const base::Name&) const override { return false; }
+    bool assetExistsByUUID(const std::string&) const override { return false; }
+    std::vector<std::tuple<std::string, std::string>> getCollection(cm::store::ResourceType) const override
+    {
+        return {};
+    }
+    std::tuple<std::string, cm::store::ResourceType> resolveNameFromUUID(const std::string&) const override
+    {
+        return {"", cm::store::ResourceType::UNDEFINED};
+    }
+    std::string resolveUUIDFromName(const std::string&, cm::store::ResourceType) const override { return {}; }
+
+protected:
+    cm::store::NamespaceId m_ns;
+};
+
+} // namespace
+
```

Comment:
Tests are missing that prove obtaining the handler in a multithreaded environment.